coalib/parsing/Globbing.py

```diff
@@ -252,14 +252,20 @@ def _absolute_flat_glob(pattern):
     return
 
 
-def _iter_relative_dirs(dirname):
+def _iter_relative_dirs(dirname, ignored_globs=None):
     """
     Recursively iterates subdirectories of all levels from dirname
 
+    :param ignored_globs: List of globs to ignore when matching files
     :param dirname: Directory name
     :return:        Iterator that yields files and directory from the given dir
                     and all it's (recursive) subdirectories
     """
+    if ignored_globs != None:
+        for ignored_dir in ignored_globs:
+            if dirname.endswith(ignored_dir):
```

Comment:
Oh, right, sorry, I didn't realize.

I think this method is the best we can do indeed, though I'm not super familiar with this code. It might turn out to be a good idea to do the ignoring elsewhere. If I were implementing this I'd just try to handle the ignoring as soon as possible and run benchmarks to make sure I'm not missing some alternative code path that won't have the ignoring.